src/main/java/com/maxmind/db/Reader.java

```diff
@@ -222,20 +279,98 @@ private int findIpV4StartNode(ByteBuffer buffer)
         return node;
     }
 
-    private int readNode(ByteBuffer buffer, int nodeNumber, int index)
-        throws InvalidDatabaseException {
+    /**
+     * Returns an iterator within the specified network.
+     * Please note that a MaxMind DB may map IPv4 networks into several locations
+     * in an IPv6 database. This iterator will iterate over all of these locations
+     * separately. To only iterate over the IPv4 networks once, use the
+     * SkipAliasedNetworks option.
+     * @param <T> Represents the data type(e.g., Map, HastMap, etc.).
+     * @param network Specifies the network to be iterated.
+     * @param skipAliasedNetworks Boolean for skipping aliased networks.
+     * @return Networks
+     * @throws BadVersionException Exception for using an IPv6 network in ipv4-only database.
+     * @throws ClosedDatabaseException Exception for a closed databased.
+     * @throws InvalidDatabaseException Exception for an invalid database.
+     */
+    public <T> Networks<T> networksWithIn(Network network, boolean skipAliasedNetworks) 
+        throws BadVersionException, ClosedDatabaseException, InvalidDatabaseException {
+        InetAddress networkAddress = network.getNetworkAddress();
+        if (this.metadata.getIpVersion() == 4 && networkAddress instanceof Inet6Address) {
+            throw new BadVersionException(networkAddress);
+        }
+
+        byte[] ipBytes = networkAddress.getAddress();
+        int prefixLength = network.getPrefixLength();
+
+        if (this.metadata.getIpVersion() == 6 && ipBytes.length == IPV4_LEN) {
+            if (skipAliasedNetworks) {
+                ipBytes = new byte[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                        ipBytes[0], ipBytes[1], ipBytes[2], ipBytes[3] };
+            } else {
+                // Convert it to the IP address (in 16-byte from) of the IPv4 address.
+                ipBytes = new byte[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                    -1, -1, // -1 is for 0xff.
+                    ipBytes[0], ipBytes[1], ipBytes[2], ipBytes[3]};
+            }
+            prefixLength += 96;
+        }
+
+        int[] traverseResult = this.traverseTree(ipBytes, 0, prefixLength);
+        int node = traverseResult[0];
+        int prefix = traverseResult[1];
+
+        Networks<T> networks = new Networks<T>(this, skipAliasedNetworks,
+            new Networks.NetworkNode[]{ new Networks.NetworkNode(ipBytes, prefix, node) });
+
+        return networks;
+    }
+
+    /**
+     * Returns the node number and the prefix for the network.
+     * @param ip The ip address to travese.
+     * @param node The number of the node.
+     * @param bitCount The prefix.
+     * @return int[]
+     */
+    public int[] traverseTree(byte[] ip, int node, int bitCount)
```

Comment:
I think this should be private. It doesn't make sense for it to be part of the public API.